src/tests/Services/CollectCmsDataServiceTest.php

```diff
@@ -0,0 +1,179 @@
+        $this->theme_proxy_mock->method('get_version')->willReturn('v.4.5');
+        $this->functions_proxy_mock = $this->createMock(FunctionsProxy::class);
+        $this->tools_helper_mock = $this->createMock(ToolsHelper::class);
+
+        $this->collect_cms_data_service = new CollectCmsDataService(
+            $this->alma_settings_mock,
+            $this->alma_logger_mock,
+            $this->payload_formatter_mock,
+            $this->security_helper_mock,
+            $this->option_proxy_mock,
+            $this->theme_proxy_mock,
+            $this->functions_proxy_mock,
+            $this->tools_helper_mock
+        );
+    }
+
+    public function test_send_url()
+    {
+        $this->alma_settings_mock->alma_client = $this->createMock(Client::class);
+        $this->alma_settings_mock->alma_client->configuration = $this->createMock(Configuration::class);
+
+        $this->tools_helper_mock->expects($this->once())
+            ->method('url_for_webhook')
+            ->with(CollectCmsDataService::COLLECT_URL)
+            ->willReturn('http://example.com/woocommerce_api_alma_collect_data_url');
+
+        $this->alma_settings_mock->alma_client->configuration->expects($this->once())
+            ->method('sendIntegrationsConfigurationsUrl')
+            ->with('http://example.com/woocommerce_api_alma_collect_data_url');
+
+        assertNull($this->collect_cms_data_service->send_url());
+    }
+
+    public function test_handle_collect_cms_data_without_signature_header()
+    {
+        $this->alma_logger_mock->expects($this->once())
+            ->method('error')
+            ->with("Header key X-Alma-Signature doesn't exist");
+
+        $this->functions_proxy_mock->expects($this->once())
+            ->method('send_http_response')
+            ->with(array('error' => 'Header key X-Alma-Signature doesn\'t exist'), 403);
+        $this->security_helper_mock->expects($this->never())->method('validate_collect_data_signature');
+
+        $this->assertNull($this->collect_cms_data_service->handle_collect_cms_data());
+    }
+
+    public function test_handle_collect_cms_data_with_invalid_signature()
+    {
+        $_SERVER['HTTP_X_ALMA_SIGNATURE'] = 'invalid_signature';
+
+        $this->security_helper_mock->expects($this->once())
+            ->method('validate_collect_data_signature')
+            ->willThrowException(new AlmaInvalidSignatureException("Invalid signature"));
+
+        $this->alma_logger_mock->expects($this->once())
+            ->method('error')
+            ->with("Invalid signature");
+
+        $this->functions_proxy_mock->expects($this->once())
+            ->method('send_http_response')
+            ->with(array('error' => 'Invalid signature'), 403);
+
+        $this->payload_formatter_mock->expects($this->never())->method('formatConfigurationPayload');
+
+        $this->assertNull($this->collect_cms_data_service->handle_collect_cms_data());
```

Comment:
I don't think we assert Null in the test, but Void

Reply:
assertNull check the void return for a fonction too. assertVoid does not exist